resource_mock/__init__.py

```diff
@@ -0,0 +1,112 @@
+
+    def _load_f2b(self, site: str) -> Dict[str,str]:
+        out_dict = {}
+        with open(f'{self.path}/gmos{site}_fpu_barcode.txt') as f:
+            
+            for row in f:
+                fpu, barcode = row.split()
+                out_dict[fpu] = barcode
+        return out_dict
+    
+    def _to_bool(self, b: str) -> bool:
+        return True if b == 'Yes' else False
+
+    def _excel_reader(self, sites: List[str]) -> NoReturn:
+        workbook = load_workbook(filename=f'{self.path}/2018B-2019A Telescope Schedules.xlsx')
+        for site in sites:
+            sheet = workbook[site]
+            self.instruments[site.lower()] = {row[0].value.strftime("%Y-%m-%d"): 
+                                              [c.value for c in row[3:] ] for row in sheet.iter_rows(min_row=2)}
+
+            self.mode[site.lower()] = {row[0].value.strftime("%Y-%m-%d"): row[1].value for row in sheet.iter_rows(min_row=2)}
+            self.LGS[site.lower()] = {row[0].value.strftime("%Y-%m-%d"): 
+                                      self._to_bool(row[2].value) for row in sheet.iter_rows(min_row=2)}
+           
+        if not self.instruments or not self.mode or not self.LGS:
+            raise Exception("Problems on reading spreadsheet...") 
+
+    def connect(self, sites: List[str]) -> NoReturn:
+        """
+        Allows the mock to load all the data locally, emulating a connection to the API.
+
+        """
+        for site in sites:
+
+            _site = 'gs' if site == 's' else 'gn'
+            self.IFU[_site]['FPU'], self.FPU[_site] = self._load_fpu('FPU',site)
+            self.IFU[_site]['FPUr'], self.FPUr[_site] = self._load_fpu('FPUr',site)
+            self.GRAT[_site] = self._load_grat(site)
+            self.fpu_to_barcode[_site] = self._load_f2b(site)
+
+        if not self.FPU or not self.FPUr or not self.GRAT:
+            raise Exception("Problems on reading files...") 
+        
+        self._excel_reader(['GS','GN'])
+
+    def night_info(self, info: str, site: str, date: str):
+
+        if info == 'fpu':
+            return self.FPU[site][date] if date in self.FPU[site] else None
+
+        elif info == 'fpur':
+            return self.FPUr[site][date] if date in self.FPUr[site] else None
+
+        elif info == 'grat':
+            return self.GRAT[site][date] if date in self.GRAT[site] else None
+
+        elif info == 'instr':
+            return self.instruments[site][date] if date in self.instruments[site] else None
+        
+        elif info == 'LGS':
+            return self.LGS[site][date] if date in self.LGS[site] else None
+        
+        elif info == 'mode':
+            return self.mode[site][date] if date in self.mode[site] else None
+        
+        elif info == 'fpu-ifu':
+            return self.IFU[site]['FPU'][date] if date in self.IFU[site]['FPU'] else None
+        elif info == 'fpur-ifu':
+            return self.IFU[site]['FPUr'][date] if date in self.IFU[site]['FPUr'] else None
```

Comment:
This could be simplified by doing:
```
info_types = { 'fpu': self.fpu[site], fpur: self.fpur[site], 'instr': self.instruments[site], 'LGS': self.lgs[site], 'mode': self.mode[site], 'fpu-ifu': self.ifu[site]['FPU'], 'fpur-ifu': self.ifu[site]['FPUr'] }
```

and then:
```
if info in info_types:
     return info_types[info][date] if date in info_types[info]
else:
     print(f"No information about {info} is stored')
     return None
```